src/baseSantry.ts

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Event } from './type';
+import * as ErrorStackParser from 'error-stack-parser';
+import { UAParser } from 'ua-parser-js';
+import { parseDSN } from './utils/parseDSN';
+import axios from 'axios';
+
+export interface Santry {
+  handleUncaughtError(error: Error): void;
+  handleUncaughtRejection(rejection: PromiseRejectionEvent): void;
+  captureError(error: Error): Event;
+  sendEvent(event: Event): void;
+  createEventFromError(event: Event, error: Error): Event;
+  addUserAgentInfo(event: Event, userAgent: string): void;
+}
+
+export type SantryClass = new (dsn: string) => Santry;
+
+export default abstract class BaseSantry implements Santry {
+  private readonly dsn?: string;
+  private readonly options?: any;
+
+  public constructor(dsn: string) {
+    this.dsn = dsn;
+  }
+
+  public abstract platform;
```

Comment:
```suggestion
  private abstract platform;
```
단순 변수라면 getter ,setter 로 하는거 어떨까요 ?

Reply:
abstract로 만들고 싶어서 이렇게 한건데 생각해보니 protected abstract platform; 으로 하는게 낫겠네용 ㅎㅎ

Reply:
이게 private 와 abstract 를 같이 못쓰더라고요 ㅠ